pkg/image/image.go

```diff
@@ -416,6 +416,11 @@ func GetTag(image string) (string, string, error) {
 		tmp := fmt.Sprintf("error removing image tag from string %s", image)
 		return "", "", errors.New(tmp)
 	}
-	justImage, _ := strings.CutSuffix(named.String(), ":"+tagged.Tag()) // should not fail, tag is guaranteed
+	justImage, _ := strings.CutSuffix(named.String(), ":"+tagged.Tag())
+	// By default, the GetTag functions returns the latest tag if the image doesn't have a tag
+	// Below checks if the image actually has this latest tag or not
+	if tagged.Tag() == "latest" && !strings.HasSuffix(image, ":latest") {
```

Comment:
Is there some edge case where this check fails even thought the image did not have any tag?

I.e. I am thinking about a situation where the user given image has the tag latest but has something after that tag (the suffix check fails). Could that be possible?

Reply:
According to the image tag spec it seems it is not possible to have something after a image tag. We should make sure our code still works if the user gives a digest, though.

Here is the specific grammar for a image reference. It is in ebnf format
`container             ::= image-name ( ":" ( tag-name | digest ) )?`

Reply:
Can you please try running the code with a digest instead of a tag, does it cause any errors?

Reply:
I wanted to confirm that image:tag@digest format is not supported. I found this in the containers/images library which confirms this. https://github.com/containers/image/blob/a362fa045dfeef8b0b29b4da4e9b446ad312e96f/docker/docker_transport.go#L97

Reply:
So supported formats should be 

1. image
2. image:tag
3. image@digest

image:tag@digest is unsupported

Reply:
done